hub/api/tests/test_api.py

```diff
@@ -691,3 +694,31 @@ def test_htypes_list():
         "binary_mask",
         "segment_mask",
     ]
+
+
+def test_hierarchical_tensors(local_ds_generator):
+    ds = local_ds_generator()
+    ds.create_tensor("x")
+    with pytest.raises(TensorAlreadyExistsError):
+        ds.create_tensor("x/y")
+    ds.create_tensor("y/x")
+
+    assert isinstance(ds.y, Dataset)
+    assert isinstance(ds.x, Tensor)
+    assert isinstance(ds.y.x, Tensor)
+
+    assert list(ds.groups) == ["y"]
+    assert list(ds.tensors) == ["x", "y/x"]
+    assert list(ds.y.tensors) == ["x"]
+    z = ds.y.create_group("z")
+    assert "z" in ds.y.groups
+
+    c = z.create_tensor("a/b/c")
+    d = z.a.b.create_group("d")
+
+    c.append(np.zeros((3, 2)))
+
+    ds = local_ds_generator()
+    c = ds.y.z.a.b.c
+    np.testing.assert_array_equal(c[0].numpy(), np.zeros((3, 2)))
+    assert "d" in ds.y.z.a.b.groups
```

Comment:
Add tests: `ds.create_tensor('y/z/a/b/d/foo')` -- i.e. create tensor in leaf group starting from root.  And, maybe `d.create_tensor('bar')` -- i.e. create from immediate parent, `a.create_tensor('b/d/baz')` -- i.e. create tensor from an ancestor group.